gtfs/__main__.py

```diff
@@ -67,109 +86,151 @@ def list_feeds(
         ),
     ] = False,
 ) -> None:
-    """Filter feeds spatially based on bounding box."""
-    if bbox is None and predicate is not None:
-        raise typer.BadParameter(
-            f"Please pass a bbox if you want to filter feeds spatially based on predicate = {predicate}!"
-        )
-    elif bbox is not None and predicate is None:
-        raise typer.BadParameter(
-            f"Please pass a predicate if you want to filter feeds spatially based on bbox = {bbox}!"
-        )
+    """Filter feeds spatially based on bounding box or search string.
+
+    :param bbox: set of coordinates to filter feeds spatially
+    :param predicate: the gtfs feed should intersect or should be contained inside the user's bbox
+    :param search: Search for feeds based on a string.
+    :param pretty: display feeds inside a pretty table
+    """
+    sources: list = feed_sources
```

Comment:
why is this copying?

Reply:
can't we use `feed_source` directly? or are you mutating it somewhere?

Reply:
yes I am mutating it later in the if condition